### ISSUE TITLE
Plugin crash in Windows 10,  "activationContext" is undefined since version 1.6.4

### ISSUE THREAD

**Issue by eduboxgithub** [phonegap/phonegap-plugin-push#1036]
### Expected Behaviour
Init the plugin and complete the initialization.

### Actual Behaviour
The plugin crashes and app force closes, when trying to get the activationContext inside the plugin source code (PushPluginProxy.js)

#### Steps to Reproduce 
Init the app with the push plugin installed and invoke the method to init the plugin.

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
Windows 10 Mobile - OS Build 10.0.14367.0

#### Cordova CLI version and cordova platform version

    cordova --version                                   6.1.1 and also 6.2
    cordova platform version windows         4.3.2

#### Plugin version

    Works with version 1.6.3 but not with 1.6.4 to 1.72.

#### Sample Code that illustrates the problem

The problem is the PushPluginProxy.js at line 70 when trying to access properties of an undefined variable. 

```
var context = cordova.require('cordova/platform').activationContext;
var launchArgs = context.args;
```

#### Logs taken while reproducing problem

Error log
```
Exception is about to be caught by JavaScript library code at line 70, column 21 in ms-appx-web://becc1407.XXXapp/www/plugins/phonegap-plugin-push/src/windows/PushPluginProxy.js

0x800a138f - JavaScript runtime error: Unable to get property 'args' of undefined or null reference
If there is a handler for this exception, the program may be safely continued.
...
Unhandled exception at line 4, column 330 in eval code
0x800a138f - JavaScript runtime error: Unable to get property 'args' of undefined or null reference
Line:  var context = cordova.require('cordova/platform').activationContext; is "undefined".
```

**Comment by rakatyal** [comment]
I am unable to reproduce this. Tried using CLI 6.1.1, windows 4.3.2 and plugin versions 1.6.4 and 1.7.1 on OS build 10.0.10586.0  
Could you try updating the OS to latest stable version? I would also try running the app on an emulator.

**Comment by napcat** [comment]
Hello @rakatyal. 
Thank you for your time on this issue.

Maybe I did not explained correctly. This only happens on a physical mobile device, probably in others devices too, but I only have one testing device a Nokia Lumia 640 LTE. If I test the app on desktop the app does not crash.

My desktop has all the latest OS updates:
Windows 10 Pro
Version: 1511
OSbuild: 10586.420

It also have the latest Visual Studio update 14.25421.03 Update 3.

My testing device is a:
Nokia Lumia 640 LTE
Windows 10 Mobile
OS build: 10.0.14372.0
Version: 1607

The device was no new updates available. If I try to run the app on this device the app crashes when trying to init the plugin (> 1.6.4 version). If I use the 1.6.3 version the app runs fine in the device.

I looked at the code and in the 1.6.3 does not exist this line:

`var context = cordova.require('cordova/platform').activationContext;`

Above the 1.6.4 that line appears and the plugins/app crashes on my device at that precise line.

**Comment by lucorn** [comment]
I'm running into the same issue both on Windows Phone and Windows Desktop build 14383 (the upcoming Window 10 Anniversary Update).

**Comment by JD76** [comment]
I have the same problem. I downgrade to version 1.6.3,  but now there is no "lauchArgs" variable defined.

**Comment by JD76** [comment]
Hi, I just updated to version 1.7.4 and now it works!

**Comment by napcat** [comment]
I updated to 1.7.4 and the app still crashes in the same line...

Is there a way, in Visual Studio to get the crash log or any other info that may help solving the problem?

**Comment by tomasvarg** [comment]
I've got the same issue, Windows 10 & Windows Phone 8.1 devices with somewhat latest update & latest versions of relevant stuff (Cordova 6.2.0, Cordova-windows 4.4.1 and PushPlugin 1.7.4); it does not matter if the app is associated with the Windows Store or not.

The context as in `var context = cordova.require('cordova/platform').activationContext` is obviously not guaranteed to be populated, so as a workaround I simply added a check if it's not empty, like changing the line
https://github.com/phonegap/phonegap-plugin-push/blob/master/src/windows/PushPluginProxy.js#L69
from
```
                    var launchArgs = context.args;
```
to
```
                    var launchArgs = context ? context.args : null;
```

I don't know much about cordova internals (why the `activationContext` is empty) nor what exactly is the code dependent on the `launchArgs` good for but at least it works (and push notifications  are being received).

**Comment by rakatyal** [comment]
Hi guys. Unfortunately I am not currently working on this plugin. @daserge: Could you please give us an estimate when these will be investigated?

**Comment by daserge** [comment]
@rakatyal, this is due to behavior change in the latest OS versions, because of which WinJS activated event is not fired in case of injecting `base.js` from code.
I'm working on a workaround for this - the issue can be avoided if you specify `<script type="text/javascript" src="/www/WinJS/js/base.js"></script>` directly in the `index.html`.

**Comment by daserge** [comment]
Workaround for https://issues.apache.org/jira/browse/CB-11658:
https://github.com/apache/cordova-windows/pull/191

**Comment by macdonst** [comment]
Released 1.8.1 to fix this issue.
